src/sentry/eventstore/models.py

```diff
@@ -591,10 +547,10 @@ def __init__(
     ):
         super().__init__(project_id, event_id, snuba_data=snuba_data)
         self.group_id = group_id
-        self.groups = groups
+        self.groups = groups or []
         self.data = data
```

Comment:
@asottile-sentry @wedamija I need help here. I can't figure out how to fix this error.

```
mypy src/sentry/eventstore/models.py
src/sentry/eventstore/models.py:549: error: Incompatible types in assignment (expression has type "Mapping[str, Any] | None", variable has type "NodeData")  [assignment]
Found 1 error in 1 file (checked 1 source file)
```

Reply:
mypy doesn't understand disparate setters unfortunately -- you can work around this by making the property setter call some private method and have this call that same private method